cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -71,6 +75,43 @@ using v1::by_default;
 
 } // namespace method
 
+namespace result_options {
+
+/// Represents result option flag
+/// Behaves like a regular :expr`enum`.
+struct result_option_id_t : public result_option_id {
+    result_option_id_t() : result_option_id{} {}
+    result_option_id_t(const result_option_id& base) : result_option_id{ base } {}
+};
+
+} // namespace result_options
```

Comment:
Why this structure is declared in the `result_options` namespace?  Nothing prevents you from moving this to the parent namespace

Reply:
Nothing prevents to keep it in this namespace either. As far as i remember it was discussed with other api changes/